pyproject.toml

```diff
@@ -30,10 +30,15 @@ classifiers = [
 dynamic = ["version", "readme"]
 
 dependencies = [
-    "botocore >=1.35.74, <1.35.89", # NOTE: When updating, always keep `project.optional-dependencies` aligned
     "aiohttp >=3.9.2, <4.0.0",
-    "wrapt >=1.10.10, <2.0.0",
     "aioitertools >=0.5.1, <1.0.0",
+    "botocore >=1.35.74, <1.35.89", # NOTE: When updating, always keep `project.optional-dependencies` aligned
+    "python-dateutil >=2.1, <3.0.0",
+    "jmespath >=0.7.1, <2.0.0",
+    "multidict >=6.0.0, <7.0.0",
```

Comment:
multdict comes in via aiohttp i believe

Reply:
Still, we use it directly (for headers processing?) and should make that dependency explicit IMO.

Reply:
only reason I think we use it is because aiohttp uses it I think (to match them). so if they changed it we'd have to charge on our side

Reply:
But until we notice and fix the dependencies on our end, our users will have to deal with broken installations.

Reply:
@webknjaz: What's your take on dependency specification?
1. Should we list everything we use directly in our codebase (i.e. in import statements) as explicit install and/or dev dependencies?
2. Or is it enough to only list those dependencies which are not already indirectly included through other dependencies: e.g. avoid multidict, since it's already required via aiohttp?

I feel strongly about 1. but cannot think of a reference that requires or recommends that approach.

Reply:
(1) is correct. All the direct deps should be tracked explicitly, regardless of their incidental presence in the dependency tree as transitive.
This is the only place where what is explicitly required is tracked, plus changes in transitive dependencies may result in a situation when a transitive dependency silently disappears from the tree, and you're left with an incomplete dependency specification and import errors to debug.